### PR TITLE
simplify scatter mean

### PR DESCRIPTION
**Pull request by CarloLucibello**

### CODE REVIEW

File: src/scatter.jl [hunk 1]
```diff
@@ -148,10 +148,22 @@ end
...
+    print("START")
+    counts = fill!(similar(Δ, Int, size(Δ)[end-M+1:end]), 0)
+    scatter!(+, counts, fill!(similar(idx), 1), idx)
+    den = gather(counts, idx)
+    
+    # make num and den broadcast compatible
+    for i in 1:ndims(num)-ndims(den)
+        den = unsqueeze(den)
+    end
+    return safe_div.(num, den)
```
Comment (DhairyaLGandhi) [inline src/scatter.jl:163]: `save_div` looks odd, is there precedent to doing this sort of division?

Comment (CarloLucibello) [reply src/scatter.jl:163]: it's already used in the scatter-mean forward pass. Here it's needed to correctly backpropagate zeros instead of infs

Comment (DhairyaLGandhi) [reply src/scatter.jl:161]: Could you link to where pytorch/ tf does this?

Comment (CarloLucibello) [reply src/scatter.jl:161]: I'm not familiar with those codebases

**Comment by CarloLucibello** [comment]
gpu tests not passing is expected, NNlibCUDA is relying on an internal function deleted here that is not needed in or in NNlibCUDA anymore. Will file a PR fixing NNlibCUDA

**Comment by CarloLucibello** [comment]
btw, stuff like this that have to coordinate NNlib and NNlibCUDA PRs have been very annoying lately, I think we should revisit #300

**Comment by DhairyaLGandhi** [comment]
Both of them would need updating anyway, and tests can be run with branches (which is the cleaner approach). Either way, best to have an answer about the division before evaluating the PR.

File: src/scatter.jl (hunk 1 above)
Comment (maxfreu) [reply src/scatter.jl:161]: Just as a side note: the most relevant pytorch code is in these directories:

https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/cpu
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

**Pull request merged**